src/chai-form-snippet.ts

```diff
@@ -0,0 +1,117 @@
+import { LitElement, html, css } from 'lit';
+import { property, customElement } from 'lit/decorators.js';
+import { ChaiForm } from './chai-form';
+
+@customElement('chai-form-snippet')
+export class ChaiFormSnippet extends LitElement {
+  @property({ type: String }) targetForm: string = '';
+  @property({ type: Boolean }) isCodeSnippedVisible: Boolean = false;
+
+  static override styles = css`
+  
+  .snippet{
+    position: relative;
+
+  }
+
+  .snippet-input{
+    width: 400px;
+    height: 201px;
+  }
+
+  .snippet-header{
+       font-size: 19px;
+    background-color: black;
+    padding: 14px;
+    border-radius: 11px;
+    color: yelo;
+    color: yellow;
+    text-shadow: none;
+    font-family: sans-serif;
+  }
+
+  .snippet-header-message
+  {
+    font-size:16px;
+  }
+
+  .snippet-copy-btn{
+        background-color: black;
+    border-radius: 6px;
+    padding:10px;
+    margin-top: 9px;
+    color: yellow;
+    font-family: sans-serif;
+  }
+  `;
+
+  constructor() {
+    super();
+    this.targetForm = ''; 
+  }
+
+  @property({ type: String }) selectedVisibilityValue: string = '';
+  @property({ type: String }) htmlSnippet: string = '';
```

Comment:
One major design issue: you're emitting _inline CSS_ for each element. What you should do instead is emit a separate CSS code block that the user can copy-paste into the site header, for example. The inline CSS really clutters up the markup and makes it hard to see what is being rendered.

Also, add a bit of indentation to the HTML snippet to make it easier to read.